pkg/kubernetes/client.go

```diff
@@ -14,8 +14,8 @@ import (
 	"github.com/openservicemesh/osm/pkg/service"
 )
 
-// NewKubernetesClient returns a new Client which means to provide access to locally-cached k8s resources
-func NewKubernetesClient(kubeClient kubernetes.Interface, meshName string, stop chan struct{}) Client {
+// NewKubernetesClient returns a new kubernetes.Controller which means to provide access to locally-cached k8s resources
+func NewKubernetesClient(kubeClient kubernetes.Interface, meshName string, stop chan struct{}) (Controller, error) {
```

Comment:
Does changing the name of this function to `NewKubernetesController` make it any more clear?

Reply:
I thought about this, but kept it as is since a few other interfaces are instantiated using `NewXXXClient`. How about I handle these renames generically in a follow-up?